plugins/org.jboss.tools.openshift.ui/src/org/jboss/tools/openshift/internal/ui/handler/OpenInWebConsoleHandler.java

```diff
@@ -67,31 +70,55 @@ public Object execute(final ExecutionEvent event) throws ExecutionException {
 		return new Status(IStatus.WARNING, OpenShiftUIActivator.PLUGIN_ID, msg);
 	}
 
-	private String getWebConsoleUrl(Connection connection, IResource resource) {
+	protected String getWebConsoleUrl(Connection connection, IResource resource) {
 		StringBuilder url = new StringBuilder(connection.getHost()).append("/console");
 		String projectName = resource == null? null : resource.getNamespace();
 		if (projectName != null) {
 			url.append("/project/").append(projectName);
 		}
 		if (resource != null && !(resource instanceof IProject)) {
-			url.append("/browse");
-			//console doesn't seem to provide anchors to reach specific items
-			//so we just open the root category for all given resources
-			if (resource instanceof IBuildConfig ) {
-				url.append("/builds/").append(resource.getName());
-			} else if ( resource instanceof IBuild) {
-				String buildConfig =  resource.getLabels().get("buildconfig");
-				url.append("/builds/").append(buildConfig).append("/").append(resource.getName());
-			} else if (resource instanceof IDeploymentConfig) {
-				url.append("/deployments/").append(resource.getName());
-			} else if (resource instanceof IPod) {
-				url.append("/pods/").append(resource.getName());
-			} else if (resource instanceof IService) {
-				url.append("/services/").append(resource.getName());
-			} else if (resource instanceof IImageStream) {
-				url.append("/images/").append(resource.getName());
-			}
+			url.append(getResourceURL(resource));
 		}
 		return url.toString();
 	}
+	
+	protected String getResourceURL(IResource resource) {
```

Comment:
actually @jcantrill's refactoring is much simpler https://github.com/jbosstools/jbosstools-openshift/pull/1278/files#diff-6567f7e23a24fc10bdeda424c7c7e76dR75